backend/oasst_backend/api/v1/login.py

```diff
@@ -0,0 +1,74 @@
+import requests
+from fastapi import APIRouter, Depends, HTTPException, Request
+from oasst_backend import auth
+from oasst_backend.api import deps
+from oasst_backend.config import Settings
+from oasst_backend.models import Account
+from oasst_shared.exceptions.oasst_api_error import OasstError, OasstErrorCode
+from oasst_shared.schemas import protocol as protocol_schema
+from sqlmodel import Session
+from starlette.status import HTTP_401_UNAUTHORIZED
+
+router = APIRouter()
+
+
+@router.get("/discord")
+def login_discord(request: Request):
+    redirect_uri = f"{get_callback_uri(request)}/discord"
+    auth_url = f"https://discord.com/api/oauth2/authorize?client_id={Settings.AUTH_DISCORD_CLIENT_ID}&redirect_uri={redirect_uri}&response_type=code&scope=identify"
+    raise HTTPException(status_code=302, headers={"location": auth_url})
+
+
+@router.get("/callback/discord", response_model=protocol_schema.Token)
+def callback_discord(
+    auth_code: str,
+    request: Request,
+    db: Session = Depends(deps.get_db),
+):
+    redirect_uri = f"{get_callback_uri(request)}/discord"
+
+    # Exchange the auth code for an access token
+    token_response = requests.post(
```

Comment:
Could we make these requests with async calls with `aiohttp` instead of requests?

Reply:
Switched